packages/main/src/plugin/documentation/documentation-service.ts

```diff
@@ -0,0 +1,247 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { inject, injectable } from 'inversify';
+
+import { DocumentationInfo, DocumentationJsonInfo } from '/@api/documentation-info.js';
+
+import { ApiSenderType } from '../api.js';
+import { Disposable } from '../types/disposable.js';
+
+@injectable()
+export class DocumentationService extends Disposable {
+  private documentation: DocumentationInfo[] = [];
+  private isInitialized = false;
+
+  constructor(
+    @inject(ApiSenderType)
+    private apiSender: ApiSenderType,
+  ) {
+    super(() => {
+      this.documentation = [];
+      this.isInitialized = false;
+    });
+  }
+
+  async fetchDocumentation(): Promise<void> {
+    try {
+      const [docsJson, tutorialsJson] = await Promise.all([
+        this.fetchJsonContent('https://podman-desktop.io/docs.json'),
+        this.fetchJsonContent('https://podman-desktop.io/tutorials.json'),
+      ]);
+
+      if (docsJson && tutorialsJson) {
+        this.documentation = this.parseDocumentationFromJson(docsJson, tutorialsJson);
+      } else {
+        throw new Error('Failed to fetch documentation JSON files');
+      }
+      this.isInitialized = true;
+    } catch (error) {
+      console.error('Failed to fetch documentation at startup:', error);
+      // Fallback to predefined documentation if fetching fails
+      this.documentation = this.getFallbackDocumentation();
+      this.isInitialized = true;
+    }
+  }
+
+  async getDocumentationItems(): Promise<DocumentationInfo[]> {
+    if (!this.isInitialized) {
+      await this.fetchDocumentation();
+    }
+    return this.documentation;
+  }
+
+  async refreshDocumentation(): Promise<void> {
+    this.isInitialized = false; // Force re-fetch
+    await this.fetchDocumentation();
+    this.apiSender.send('documentation-updated');
+  }
+
+  private async fetchJsonContent(url: string): Promise<Array<{ name: string; url: string }> | null> {
```

Comment:
we do not use `null`, eslint is happy about that??

Reply:
let's avoid returning null, either return something valid or throw an error, let's avoid the caller to handler yet another error scenario